### PR TITLE
CEJST Map 

### PR DESCRIPTION
**Pull request by NatHillardUSDS**
Addresses issue #100 from the FE - for now we're using the census-derived tile server until we get our local one set up. 

Work in progress for now. To fix:
- [ ] Style layer
- [ ] Make layer switcher functional (can use dummy data in the meantime)
- [ ] Remove J40Aside and fix sizing
- [ ] Adopt grid for spacing
- [ ] Tests

### CODE REVIEW

File: client/.eslintrc.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ module.exports = {
     '@typescript-eslint',
   ],
   'rules': {
+    'max-len': [2, {'code': 120, 'tabWidth': 4, 'ignoreUrls': true}],
```
Comment (TomNUSDS) [inline client/.eslintrc.js:25]: Ha! It's 2021... I personally think 120 is fine.

File: client/src/components/HowYouCanHelp.module.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.howYouCanHelpContainer {
```
Comment (TomNUSDS) [inline client/src/components/HowYouCanHelp.module.scss:1]: Interesting. Should we start slitting out stuff in `styles/globals.scss` and moving them into components?

File: client/src/components/map.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        new TileLayer({
+          source: new XYZ({
+            url: 'https://{1-4}.basemaps.cartocdn.com/light_all/{z}/{x}/{y}@2x.png',
+          }),
+        }),
+        initalFeaturesLayer,
+      ],
+      view: new View({
+        projection: 'EPSG:3857',
+        center: fromLonLat([-95.7129, 37.0902]), // [37.0902, 95.7129],
```
Comment (TomNUSDS) [inline client/src/components/map.tsx:51]: Maybe an issue to remember to change this back?

File: client/src/components/map.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+// https://taylor.callsen.me/using-openlayers-with-react-functional-components/
+const MapWrapper = ({features}: IMapWrapperProps) => {
+  const [map, setMap] = useState<Map>();
+  const [featuresLayer, setFeaturesLayer] = useState<VectorLayer>();
+
+  const mapElement = useRef() as
+        React.MutableRefObject<HTMLInputElement>;
+
+  useEffect( () => {
+    const initalFeaturesLayer = new VectorTileLayer({
```
Comment (TomNUSDS) [inline client/src/components/map.tsx:31]: spelling `initial`

File: client/src/pages/cejst.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+          Explore the map below or learn
+          more about the methodology and data indicators used to
+          prioritize Justice40 communities.
+        </p>
+        <Alert
+          type="warning"
+          heading="Limited Data Sources">
+          <p>
+            In this tool, we are using data sources that our
+            combined by our cummulative impact methodology.
```
Comment (TomNUSDS) [inline client/src/pages/cejst.tsx:45]: sp: cumulative

File: client/src/pages/cejst.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+            Our sources were selected because sit amet,
```
Comment (TomNUSDS) [inline client/src/pages/cejst.tsx:46]: `sit amet, consectetur adipiscin` 🥇

File: client/src/pages/timeline.tsx [hunk 7]
```diff
@@ -6,7 +6,7 @@ import renewIcon from '/node_modules/uswds/dist/img/usa-icons/autorenew.svg';
 
 interface TimelinePageProps {
   location: Location;
-}
+};
```
Comment (TomNUSDS) [inline client/src/pages/timeline.tsx:9]: `;` not needed.

**Review by TomNUSDS — APPROVED** [review]
Nothing critical, minor comments/observations. Approved. 🚀

File: client/src/components/map.tsx [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+      view: new View({
+        projection: 'EPSG:3857',
+        center: fromLonLat([-95.7129, 37.0902]), // [37.0902, 95.7129],
+        zoom: 3,
+      }),
+      controls: [],
+    });
+
+    setMap(initialMap);
+    setFeaturesLayer(initalFeaturesLayer);
```
Comment (TomNUSDS) [inline client/src/components/map.tsx:58]: This line is giving me a typescript type warning: 
`
TS2345: Argument of type 'VectorTileLayer' is not assignable to parameter of type 'SetStateAction<VectorLayer | undefined>'.   Type 'VectorTileLayer' is not assignable to type 'VectorLayer'.     Property 'createRenderer' is protected in type 'VectorTileLayer' but public in type 'VectorLayer'
`

**Review by TomNUSDS — CHANGES_REQUESTED** [review]
Investigate a typescript warning on map.tsx:58

File: client/src/components/HowYouCanHelp.module.scss (hunk 2 above)
Comment (NatHillardUSDS) [reply client/src/components/HowYouCanHelp.module.scss:1]: We chatted a bit about this in Slack, but posting here for future reference also: 
To generate these, until the gatsby-plugin-scss-typescript plugin is working again (see more [here](https://github.com/usds/justice40-tool/pull/69)) I ended up using `gatsby-plugin-scss-typescript` to generate the files and then swapping it back for `gatsby-plugin-sass`

File: client/src/components/map.tsx (hunk 3 above)
Comment (NatHillardUSDS) [reply client/src/components/map.tsx:51]: ah, we can just remove this actually

File: client/src/components/map.tsx (hunk 4 above)
Comment (NatHillardUSDS) [reply client/src/components/map.tsx:31]: d'oh - thanks!

File: client/src/pages/cejst.tsx (hunk 5 above)
Comment (NatHillardUSDS) [reply client/src/pages/cejst.tsx:45]: ah, this was in the copy of the design; I can comment over there also

File: client/src/pages/cejst.tsx (hunk 6 above)
Comment (NatHillardUSDS) [reply client/src/pages/cejst.tsx:46]: yeah we should get permanent text - left a comment just now!

File: client/src/pages/timeline.tsx (hunk 7 above)
Comment (NatHillardUSDS) [reply client/src/pages/timeline.tsx:9]: good callout!

**Review by TomNUSDS — APPROVED** [review]
lgtm

**Comment by NatHillardUSDS** [comment]
Will address above list items in separate PR

**Pull request merged**